handlers/api.go

```diff
@@ -839,6 +848,177 @@ func getSyncCommitteeFor(validators []uint64, period uint64) ([]interface{}, err
 	return utils.SqlRowsToJSON(rows)
 }
 
+func getSyncCommitteeStatistics(validators []uint64, epoch uint64) (*SyncCommitteesStats, error) {
+	r := SyncCommitteesStats{}
+
+	if epoch < utils.Config.Chain.Config.AltairForkEpoch {
+		// no sync committee duties before altair fork
+		return &r, nil
+	}
+
+	if len(validators) == 0 {
+		// no validators mean no sync committee duties either
+		return &r, nil
+	}
+
+	////
+	// calculate expected slots
+
+	// retrieve activation and exit epochs from database per validator
+	var validatorsEpochInfo = []struct {
+		Id              int64  `db:"validatorindex"`
+		ActivationEpoch uint64 `db:"activationepoch"`
+		ExitEpoch       uint64 `db:"exitepoch"`
+	}{}
+
+	query, args, err := sqlx.In(`SELECT validatorindex, activationepoch, exitepoch FROM validators WHERE validatorindex IN (?) ORDER BY validatorindex ASC`, validators)
+	if err != nil {
+		return nil, err
+	}
+	err = db.ReaderDb.Select(&validatorsEpochInfo, db.ReaderDb.Rebind(query), args...)
+	if err != nil {
+		return nil, err
+	}
+
+	// we need all related and unique timeframes (activation and exit sync period) for all validators
+	const noExitEpoch = uint64(9223372036854775807)
+	uniquePeriods := make(map[uint64]bool)
+	uniquePeriods[utils.SyncPeriodOfEpoch(epoch)] = true
+	for _, v := range validatorsEpochInfo {
+		// activation epoch
+		firstSyncEpoch := v.ActivationEpoch
+		if utils.Config.Chain.Config.AltairForkEpoch > v.ActivationEpoch {
+			firstSyncEpoch = utils.Config.Chain.Config.AltairForkEpoch
+		}
+		uniquePeriods[utils.SyncPeriodOfEpoch(firstSyncEpoch)] = true
+
+		// exit epoch (if any)
+		if v.ExitEpoch != noExitEpoch && v.ExitEpoch > firstSyncEpoch {
+			uniquePeriods[utils.SyncPeriodOfEpoch(v.ExitEpoch)] = true
+		}
+	}
+
+	// transform map to slice; this will be used to query sync_committees_count_per_validator
+	timeStampSlice := make([]uint64, 0, len(uniquePeriods))
```

Comment:
Isn't this a period slice not timestamp?

Reply:
You are right, changed.